core/src/main/java/org/apache/calcite/avatica/remote/AvaticaCommonsHttpClientImpl.java

```diff
@@ -147,10 +147,10 @@ protected void configureHttpsRegistry(RegistryBuilder<ConnectionSocketFactory> r
 
   private SSLContext getSSLContext() throws Exception {
     SSLContextBuilder sslContextBuilder = SSLContexts.custom();
-    if (null != truststore && null != truststorePassword) {
+    if (null != truststore) {
       loadTrustStore(sslContextBuilder);
```

Comment:
I really shouldn't have opened this particular can of worms :)

As it turns out, there is no way to handle (or at least create) unencrypted private keys and keystores with Java. What I thought were unencrypted keys and keystores are in fact simply keys / keystores with the empty string as password.

The most user-friendly thing would probably be to set the default for the key*password parameters to empty string instead of null, but for now I have just reverted the change.

Reply:
> The most user-friendly thing would probably be to set the default for the key*password parameters to empty string instead of null, but for now I have just reverted the change.

👍 that works for me.